### ISSUE TITLE
Make type definitions available through TSD installation

### ISSUE THREAD

**Issue by tedvanderveen** [spira/angular-jwt-auth#35]
Really like this library and we would like to replace angular-jwt with this one.
The only thing that seems to be missing is a reference to the type definitions file through definitelytyped and thereby the ability to streamline the installation using TSD commands.
Please add these to https://github.com/borisyankov/DefinitelyTyped
http://definitelytyped.org/tsd/

**Comment by zakhenry** [comment]
@tedvanderveen thanks, I'll look into it, though I'm not sure how TSD accepts contributed code as there doesn't seem to be any pull requests with new definitions there.

It the meantime the `.d.ts` file is distributed with bower so you can directly reference the file.

**Comment by tedvanderveen** [comment]
@xiphiaz please refer to http://definitelytyped.org/guides/contributing.html
Should be a quick process :-)
Thanks!

**Comment by tedvanderveen** [comment]
@xiphiaz Just like in your own code, we use a central "typings" folder that holds all external type definitions and this folder is maintained using TSD. So using bower and reference the .d.ts file from outside the typings folder would break our setup.

**Comment by zakhenry** [comment]
@tedvanderveen I am intending to add angular-jwt-auth to TSD when I find some time, but in the meantime you can actually have bower referenced `d.ts` files co-exist with typings folder - I use this pattern in spira/spira.

The local typings file can automatically generate the linkings to dependencies with the `tsd link` command.

Note that I had to do a bit of a hack to stop dependency collisions however - you can see it in https://github.com/spira/spira/blob/master/gulpfile.js#L115

Basically what the spira gulpfile does is copy all the `d.ts` files from both local typings and bower dependencies before it compiles against them.

As I mentioned, I do intend to add this package to definitely typed, but I don't currently have the time - I'll have to restructure the distributed file slightly, and add tests.

**Comment by tedvanderveen** [comment]
@xiphiaz cool, thanks for this. Please give me shout when you found time to submit to TSD or if I can support you with anything.

**Comment by tedvanderveen** [comment]
@xiphiaz did you have a chance to look into this? Thanks

**Comment by zakhenry** [comment]
@tedvanderveen I'm finally able to give this library some love - see https://github.com/spira/angular-jwt-auth/pull/44 - I am refactoring to use commonjs pattern for standard loading with typescript, and be typings compatible.
Note this will be a major version bump as I am deprecating bower support, and going npm-only per common typescript conventions now. The API should not change at all though.
ETA end of this week.

**Comment by tedvanderveen** [comment]
@xiphiaz Nice!

**Comment by zakhenry** [comment]
Closed in #50
